lib/nostrum/api.ex

```diff
@@ -486,107 +487,119 @@ defmodule Nostrum.Api do
+  Nostrum.Api.get_channel(381889573426429952)
+  {:ok, %Nostrum.Struct.Channel{id: 381889573426429952}}
+  ```
+  """
+  @spec get_channel(Channel.id()) :: error | {:ok, Channel.t()}
+  def get_channel(channel_id) when is_snowflake(channel_id) do
+    request(:get, Constants.channel(channel_id))
+    |> handle_request_with_decode({:struct, Channel})
+  end
 
-  Raises `Nostrum.Error.ApiError` if error occurs while making the rest call.
+  @doc ~S"""
+  Same as `get_channel/1`, but raises `Nostrum.Error.ApiError` in case of failure.
   """
-  @spec get_channel!(integer) :: no_return | Nostrum.Struct.Channel.t()
+  @spec get_channel!(Channel.id()) :: no_return | Channel.t()
   def get_channel!(channel_id) do
     get_channel(channel_id)
     |> bangify
   end
 
-  @doc """
-  Edit a channel.
+  @doc ~S"""
+  Modifies a channel's settings.
 
-  Edits a channel with `options`
+  If a `t:Nostrum.Struct.Channel.guild_channel/0` is being modified, this
+  endpoint requires the `MANAGE_CHANNEL` permission. It fires a
+  `t:Nostrum.Consumer.channel_update/0` event. If a
+  `t:Nostrum.Struct.Channel.channel_category/0` is being modified, then this
+  endpoint fires multiple `t:Nostrum.Consumer.channel_update/0` events.
 
-  `options` is a kwl with the following optional keys:
-   * `name` - New name of the channel.
-   * `position` - Position of the channel.
-   * `topic` - Topic of the channel. *Text Channels only*
-   * `bitrate` - Bitrate of the voice channel. *Voice Channels only*
-   * `user_limit` - User limit of the channel. 0 for no limit. *Voice Channels only*
-  """
-  @spec edit_channel(
-          integer,
-          name: String.t(),
-          position: integer,
-          topic: String.t(),
-          bitrate: String.t(),
-          user_limit: integer
-        ) :: error | {:ok, Nostrum.Struct.Channel.t()}
-  def edit_channel(channel_id, options) do
-    case request(:patch, Constants.channel(channel_id), options) do
-      {:ok, body} ->
-        {:ok, Poison.decode!(body)}
+  If successful, returns `{:ok, channel}`. Otherwise, returns a `t:Nostrum.Api.error/0`.
 
-      other ->
-        other
-    end
-  end
+  ## Options
 
-  @doc """
-  Edit a channel.
+    * `:name` (string) - 2-100 character channel name
+    * `:position` (integer) - the position of the channel in the left-hand listing
+    * `:topic` (string) (`t:Nostrum.Struct.Channel.text_channel/0` only) -
+    0-1024 character channel topic
+    * `:nsfw` (boolean) (`t:Nostrum.Struct.Channel.text_channel/0` only) -
+    if the channel is nsfw
+    * `:bitrate` (integer) (`t:Nostrum.Struct.Channel.voice_channel/0` only) -
+    the bitrate (in bits) of the voice channel; 8000 to 96000 (128000 for VIP servers)
+    * `:user_limit` (integer) (`t:Nostrum.Struct.Channel.voice_channel/0` only) -
+    the user limit of the voice channel; 0 refers to no limit, 1 to 99 refers to a user limit
+    * `:permission_overwrites` (list of `t:Nostrum.Struct.Overwrite.t/0`) - channel
```

Comment:
I think it's a little weird to say this has to be a struct, but I think it's fine as a general precaution. Maybe we could modify it to be something like `list of 't:Nostrum.Struct.Overwrite.t/0', or an equivalent map`?

Reply:
I think this is also relevant in the `create_guild_channel` method

Reply:
I'm down for this change, but I have an alternate solution we could try (one that is more struct heavy):

To solve this problem, could a `new/4` function be feasible? Might simplify the process a little bit:

```elixir
# @spec new(Overwrite.id(), Overwrite.name(), Overwrite.allow(), Overwrite.deny())

iex> alias Nostrum.Struct.Overwrite
...> ow = Overwrite.new(89918932789497856, "member", 0, 0x00001)
...> Nostrum.Api.modify_channel(381889573426429952, permission_overwrites: [ow])
{:ok, %Nostrum.Struct.Channel{}}
```

Alternatively, we could also try making builder functions for the Overwrite module (like Embed). But, I don't think this is a really elegant solution.

Reply:
`new` sounds very OOP, but for modules with permissions maybe we can think about doing something like this. We can save it for later when we do other related work. For now I think the doc change would be sufficient.

Reply:
Fixed.